x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/LifecycleMetadataService.java

```diff
@@ -54,38 +55,31 @@
 import static org.elasticsearch.xpack.core.ilm.PhaseCacheManagement.updateIndicesForPolicy;
 import static org.elasticsearch.xpack.core.searchablesnapshots.SearchableSnapshotsConstants.SEARCHABLE_SNAPSHOT_FEATURE;
 
-public class PutLifecycleMetadataService {
+public class LifecycleMetadataService {
 
-    private static final Logger logger = LogManager.getLogger(PutLifecycleMetadataService.class);
+    private static final Logger logger = LogManager.getLogger(LifecycleMetadataService.class);
 
-    private final ClusterService clusterService;
     private final NamedXContentRegistry xContentRegistry;
     private final Client client;
     private final XPackLicenseState licenseState;
     private final ThreadPool threadPool;
     private final ProjectResolver projectResolver;
-    private final MasterServiceTaskQueue<UpdateLifecyclePolicyTask> taskQueue;
+    private final MasterServiceTaskQueue<AckedClusterStateUpdateTask> taskQueue;
```

Comment:
Can we add a comment explaining that we use this queue for both additions and deletions? I don't think that's very common, but I like the idea as it avoids duplicate task queues.